src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,38 @@
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+interface Props {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  onDelete?: (todoId: number) => void;
+  loadingTodos: number[];
+  onToggle?: (todo: Todo) => void;
+  onRename?: (todo: Todo) => void;
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  onDelete,
+  loadingTodos,
+  onToggle,
+  onRename,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <TodoItem
+          key={todo.id}
+          todo={todo}
+          onDelete={onDelete}
+          isLoading={loadingTodos.includes(todo.id)}
```

Comment:
[Not critical]
The logic of detecting if a todo is loading (`loadingTodos.includes(...)`) is used in multiple places. So you can define it as a separate function and call it from all these places.